introspection-engine/connectors/mongodb-introspection-connector/src/sampler/statistics.rs

```diff
@@ -0,0 +1,339 @@
+use std::{
+    borrow::Cow,
+    cmp::Ordering,
+    collections::{BTreeMap, HashMap},
+    fmt,
+    ops::Deref,
+};
+
+use bson::Document;
+use datamodel::{
+    Datamodel, DefaultValue, Field, IndexDefinition, IndexType, Model, NativeTypeInstance, PrimaryKeyDefinition,
+    ScalarField, ScalarType, ValueGenerator, WithDatabaseName,
+};
+use introspection_connector::Warning;
+use mongodb::IndexModel;
+use native_types::MongoDbType;
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+use super::field_type::FieldType;
+
+static RESERVED_NAMES: &[&str] = &["PrismaClient"];
+
+static RE_START: Lazy<Regex> = Lazy::new(|| Regex::new("^[^a-zA-Z]+").unwrap());
+static RE: Lazy<Regex> = Lazy::new(|| Regex::new("[^_a-zA-Z0-9]").unwrap());
```

Comment:
These two statics look a bit lonely here, I think they're only used in `sanitize_string()`, can they be moved inside there?

Reply:
Will do.